src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -306,6 +306,9 @@ private class TaskLauncherActor(
           logger.info(s"Did not match offer ${offer.getId} for run spec ${runSpec.id}, ${runSpec.version}.")
           offerMatchStatistics.offer(OfferMatchStatistics.MatchResult(notMatched))
           promise.trySuccess(MatchedInstanceOps.noMatch(offer.getId))
+        case notMatched: OfferMatchResult.NoScheduledInstances =>
```

Comment:
We should theoretically never get here since `shouldLaunchInstances` [here](https://github.com/mesosphere/marathon/pull/6664/files#diff-f91c767f12c6ca4b33353f65393f692dR290) already checks for `scheduledInstances.nonEmpty`. `NonEmptyList` like a good idea.

Reply:
Yep. It would be a bug. I'd use [cats](https://typelevel.org/cats/)'s `NonEmptyList`. That opens a door we can never close 🙈.

Reply:
I implemented a simple `NonEmptyIterable`.